docs/configuration/http_filters/rate_limit_filter.rst

```diff
@@ -60,6 +60,32 @@ The following descriptors are sent:
 
 <local service cluster> is derived from the :option:`--service-cluster` option.
 
+Request Headers
+^^^^^^^^^^^^^^^
+
+.. code-block:: json
+
+  {
+    "type": "request_headers",
+    "header_name": "...",
+    "descriptor_key" : "..."
+  }
+
+header_name
+  *(required, string)* The header name to be queried from the request header and used to
+  populate the descriptor value for the *descriptor_key*.
+
+descriptor_key
+  *(required, string)* The key to use in the descriptor.
+
+The following descriptor is sent when a header contains a key that matches the *header_name*:
+
+  * ("<descriptor_key>", "<header_value_queried_from_header>")
+
+If *rate_limit_key* is set in the :ref:`route <config_http_conn_man_route_table_route_rate_limit>`, the following descriptor is sent:
+
+  * ("rate_limit", "<rate_limit_key>"), ("<descriptor_key>", "<header_value_queried_from_header>")
```

Comment:
I think sending "rate_limit" is confusing. I would maybe do "route_key" (and of course update code below)